doc/source/cluster/ray-client.rst

```diff
@@ -48,6 +50,55 @@ You can also connect using the ClientBuilder API, but this will eventually be de
    do_work.remote(2)
    #....
 
+Client arguments
+----------------
+
+Ray client is used when the address passed into ``ray.init`` is prefixed with the name of a protocol and ``://``. **This will only work with Ray 1.5+.** If you're using an older version of ray, see the ClientBuilder API section for information on passing arguments. Currently, two protocols are built into ray by default:
+
+**Local mode** is used when the address is prefixed with ``local://``, i.e. ``ray.init("local://")``. In local mode, a local cluster is created and connected to by Ray Client. Local mode accepts all of the same keywords as ``ray.init`` and uses them to configure the local cluster.
```

Comment:
What happens if I call ‘ray start —head’ on local machine and then call ‘ray.init(“ray://localhost:6379”)’ do I get the same outcome of directly doing ‘ray.init(“local://“)’?

also what happens if I do ‘ray.init(“local”)’ i.e. without the “://“?

Reply:
It looks like ray.init("local://") actually attaches a driver directly (no grpc in between) after it creates the cluster (ray.init("local://") is basically the same thing as ray.init() without any args). `ray start --head` + `ray.init("ray.init://...")` will use ray client (grpc in between). Updated the docs to reflect that "local://" attaches a driver instead of using grpc.

ray.init("local") without the "://" will result in a socketerorr, since ray.init will try to treat it like a regular address if it doesn't see a "://". Added a section near the bottom mentioning this.